python/love/producer/love_producer_csc.py

```diff
@@ -0,0 +1,552 @@
+
+        Returns
+        -------
+        `dict`
+            Heartbeat message.
+        """
+        return dict(
+            category="event",
+            producer_snd=0,
+            data=[
+                dict(
+                    csc="Heartbeat",
+                    salindex=0,
+                    data=dict(
+                        stream=dict(
+                            csc=self.remote.salinfo.name,
+                            salindex=self.remote.salinfo.index,
+                            lost=0,
+                            last_heartbeat_timestamp=-1,
+                            max_lost_heartbeats=self.heartbeat_max_lost,
+                        )
+                    ),
+                ),
+            ],
+        )
+
+    async def _handle_heartbeat(self, heartbeat_message: dict) -> Tuple[int, float]:
+        """Handler heartbeat reception.
+
+        Parameters
+        ----------
+        heartbeat_message : `dict`
+            Heartbeat message dictionary.
+
+        Returns
+        -------
+        heartbeat_lost : `int`
+            Number of heartbeat losts.
+        last_heartbeat_timestamp : `float`
+            Timestamp of the last hearbeat.
+        """
+        heartbeat_lost = heartbeat_message["data"][0]["data"]["stream"]["lost"]
+        last_heartbeat_timestamp = heartbeat_message["data"][0]["data"]["stream"][
+            "last_heartbeat_timestamp"
+        ]
+
+        try:
+            await self.remote.evt_heartbeat.next(
+                flush=True, timeout=self.heartbeat_timeout
+            )
+            heartbeat_lost = 0
+            last_heartbeat_timestamp = datetime.datetime.now().timestamp()
+        except asyncio.TimeoutError:
+            self.log.debug("Timeout waiting for hearbeat.")
+            heartbeat_lost += 1
+        except Exception as e:
+            self.log.exception("Error waiting for heartbeat.")
+            raise e
+
+        return heartbeat_lost, last_heartbeat_timestamp
+
+    def _convert_data_to_dict(self, data: Any) -> Tuple[str, dict]:
+        """Convert SalObj topic data to dictionary.
+
+        Parameters
+        ----------
+        data:
+            SalObj topic data to convert to dictionary.
+
+        Returns
+        -------
+        name: `str`
+            Assigned name of the kind of data stream.
+        data_as_dict: `dict`
+            Dictionary with the data payload.
+
+        """
+        topic_attribute_name = self.get_topic_attribute_name(data.private_revCode)
+        _, topic_name = topic_attribute_name.split("_", maxsplit=1)
+
+        data_stream = copy.deepcopy(
+            self._template_manager_message[topic_attribute_name]
+        )
+        data_vars = data.get_vars()
+
+        for topic_attribute in data_vars:
+            data_stream[topic_attribute]["value"] = data_vars[topic_attribute]
+
+        payload = (
+            data_stream
+            if topic_attribute_name in self.periodic_data
```

Comment:
How does this work?

Reply:
uhm.. I am not 100% sure what you are asking.. The construct here is:

```
value = result_1 if condition else result_2
```

it is equivalent to doing:

```
if condition:
    value = result_1
else:
    value = result_2
```

Reply:
My question here is: why the payload can be `data_stream` or a an array of only one element which is `data_stream`?

Reply:
Ah, I see.. It's because the telemetry and event messages have different data structures. For events the manages expects "data" to be an array with a dictionary and telemetry it expects "data" to be a dictionary.

See:

https://github.com/lsst-ts/LOVE-producer/blob/fcc40209ff903529230913781af22b4fdf801f08/python/love/producer/telemetries/producer.py#L80

and 

https://github.com/lsst-ts/LOVE-producer/blob/fcc40209ff903529230913781af22b4fdf801f08/python/love/producer/events/producer.py#L273

We should probably fix that in the future though, and unify them.

Reply:
Oh I see, that's true and yes, we should refactor that. Thanks for the explanation!